### PR TITLE
adds opentelemetry configs & container

### PR DESCRIPTION
**Pull request by haninator**
This adds the configuration for the Opentelemetry Collector, Prometheus and the needed config inside the docker compose file.
This also adds a Prometheus, an Opentelemetry Collector and a Jaeger Container to the docker-compose

- - -
This change is [Reviewable](https://reviewable.io/reviews/dqualizer/dqualizer/30)

### CODE REVIEW

File: config/otel-collector-config.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+processors:
+  batch:
+
+extensions:
+  health_check:
+  pprof:
+    endpoint: :1888
+  zpages:
+    endpoint: :55679
```
Comment (ArianeKraus) [inline config/otel-collector-config.yaml:25]: I can think of a use case for using health check but do we have a use case for using pprof and zpages?
If not I would remove both extensions in order to keep the overall configuration at a minimum.

File: config/prometheus.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+global:
+  scrape_interval: 15s
+  evaluation_interval: 15s
+
+scrape_configs:
+  - job_name: 'otel-collector'
+    scrape_interval: 10s
+    static_configs:
+      - targets: [ 'otel-collector:8889' ]
```
Comment (ArianeKraus) [inline config/prometheus.yaml:9]: Can we add a second scrape endpoint for otel-collector:8888?

File: docker-compose.yml [hunk 3]
```diff
@@ -107,3 +107,39 @@ services:
...
+      - ./config/prometheus.yaml:/etc/prometheus/prometheus.yml
+    ports:
+      - "9090:9090"
+
+  jaeger:
+    image: jaegertracing/all-in-one:1.45
+    restart: always
+    ports:
+      - "16686:16686"
+      - "14268"
```
Comment (ArianeKraus) [inline docker-compose.yml:125]: unless we're using it I would suggest to remote 14268 port

File: docker-compose.yml [hunk 4]
```diff
@@ -107,3 +107,39 @@ services:
...
+      - "14250"
+
+  otel-collector:
+    image: otel/opentelemetry-collector:0.67.0
+    restart: always
+    command: [ "--config=/etc/otel-collector-config.yaml" ]
+    volumes:
+      - ./config/otel-collector-config.yaml:/etc/otel-collector-config.yaml
+    ports:
+      - "1888:1888"   # pprof extension
```
Comment (ArianeKraus) [inline docker-compose.yml:135]: reminder to remove ports if we end up not using the extensions

File: docker-compose.yml [hunk 5]
```diff
@@ -107,3 +107,39 @@ services:
...
+      - "8888:8888"   # Prometheus' metrics exposed by the collector
+      - "8889:8889"   # Prometheus exporter metrics
+      - "13133:13133" # health_check extension
+      - "4317:4317"   # OTLP gRPC receiver
```
Comment (ArianeKraus) [inline docker-compose.yml:139]: I would keep the 4317 port internal, as it is currently enough for our use case

**Review by ArianeKraus — CHANGES_REQUESTED** [review]

**Review by ArianeKraus — APPROVED** [review]

**Review by franksn90 — COMMENTED** [review]
Full runthrough executed successfully

**Pull request merged**